### DISCUSSION TITLE
How to build in mqtt username and password

### DISCUSSION THREAD

**Discussion by KamenRiderKuuga** [emqx/emqx#13329] [Q&A]
I have used the pre-set MQTT account based on HTTP and database, but I actually need a way that does not rely on any external service and database files. I want to ask if there is a configuration file to achieve this.

**Comment by ieQu1** [comment]
EMQX can store usernames and passwords in the builtin database. It supports data import from file:
https://docs.emqx.com/en/emqx/latest/access-control/authn/user_management.html#importing-users

**Comment by zmstone** [comment]
v5.7.2 will allow bootstraping built-in database from a file.
see more details here: https://github.com/emqx/emqx/pull/13336
